app/models/assignment.rb

```diff
@@ -91,7 +91,15 @@ def extension
   end
 
   def notify!
-    Mumukit::Nuntius.notify! 'submissions', to_resource_h unless Organization.silenced?
+    unless Organization.silenced?
+      contexts = notification_contexts
+      update_missplaced!(notification_contexts.size > 1)
+      contexts.uniq.each { |it| Mumukit::Nuntius.notify! 'submissions', to_resource_h(context: it) }
+    end
+  end
+
+  def notification_contexts
```

Comment:
I should rename this to `current_notification_contexts` just for naming consistency